src/content/docs/tutorials/crates-tui/app_handle_action.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Handle Actions
+---
+
+In `handle_action`, we map the `Action`s back to the `app`'s corresponding method:
+
+```rust
+{{#include @code/crates-tui-tutorial-app/src/app.rs:app_handle_action}}
+```
+
+While this may seem like a lot more boilerplate at first, using an `Action` enum this way has a few
+advantages.
+
+Firstly, they can be mapped from keypresses programmatically. For example, you can define a
+configuration file that reads which keys are mapped to which `Action` like so:
+
+```toml
+[keyconfig]
+"q" = "Quit"
+"j" = "ScrollDown"
+"k" = "ScrollUp"
+```
+
+Then you can add a new `keyconfig` in the `App` like so:
+
+```rust
+struct App {
+  ...
+  // new field
+  keyconfig: HashMap<KeyCode, Action>
+}
+```
+
+If you populate `keyconfig` with the contents of a user provided `toml` file, then you can figure
+out which action to take directly from the keyconfig struct:
+
+```rust
+fn handle_event(&mut self, event: Event) -> Option<Action> {
+  if let Event::Key(key) = event {
+    return self.keyconfig.get(key.code)
+  };
+  None
+}
+```
+
+But more pertinently for us, because the `run` method has the following block of code:
+
+```rust
+while let Ok(action) = self.rx.try_recv() {
+    self.handle_action(action.clone(), &mut tui)?
+        .map(|action| self.tx.send(action));
```

Comment:
I'd prefer to push these calls down to the methods that need to raise actions - e.g. all the way into the scroll or submit_search method. We're using a "Multiple Producer..." channel for a reason - and that's that there should be multiple places that can send the messages.

Reply:
Yup! Makes sense. This `.map(|action| ...)` is vestigial code from the refactor to make things simpler.

Reply:
I've removed it now.